bundles/http/Citrix/Canvas-LMS/0b9f4444-56d6-4b4e-b108-5624645e71d2.js

```diff
@@ -1,449 +1,398 @@
-    dataStore.deleteById("enrollments", actionParameters.enrollmentId);
-  } else {
-    throw new Error(
-      `Could not reject invitation: (${response.status}: ${response.statusText})`
-    );
-  }
+async function rejectInvitation({ dataStore, client, actionParameters }) {
+	const response = await client.fetch(
+		`/api/v1/courses/${actionParameters.courseId}/enrollments/${actionParameters.enrollmentId}/reject`,
+		{
+			method: 'POST'
+		}
+	);
+	if (response.ok) {
+		dataStore.deleteById('enrollments', actionParameters.enrollmentId);
+	} else {
+		throw new Error(`Could not reject invitation: (${response.status}: ${response.statusText})`);
+	}
 }
 
 //SA-Create Invitation
-async function createInvitation({
-  dataStore,
-  client,
-  actionParameters,
-  serviceClient,
-}) {
-  const response = await client.fetch(
-    `/api/v1/courses/${actionParameters.courseId}/enrollments?enrollment[type]=StudentEnrollment&enrollment[user_id]=${actionParameters.userId}`,
-    {
-      method: "POST",
-    }
-  );
-  if (response.ok) {
-    return updateAction(
-      dataStore,
-      serviceClient,
-      actionParameters.courseId,
-      now,
-      end
-    );
-  } else {
-    throw new Error(
-      `Could not create course registration: (${response.status}: ${response.statusText})`
-    );
-  }
+async function createInvitation({ dataStore, client, actionParameters, serviceClient }) {
+	const response = await client.fetch(
+		`/api/v1/courses/${actionParameters.courseId}/enrollments?enrollment[type]=StudentEnrollment&enrollment[user_id]=${actionParameters.userId}`,
+		{
+			method: 'POST'
+		}
+	);
+	if (response.ok) {
+		return updateAction(dataStore, serviceClient, actionParameters.courseId, now, end, true);
+	} else {
+		throw new Error(`Could not create course registration: (${response.status}: ${response.statusText})`);
+	}
 }
 
 // Announcement & Enrollment API call
-async function updateAction(dataStore, serviceClient, courseId, StartDate, endDate) {
-  let announcementIncrement = 1;
-  let announcementPageExists;
+async function updateAction(dataStore, serviceClient, courseId, StartDate, endDate, isUpdateAfterAction) {
+	let announcementIncrement = 1;
+	let announcementPageExists;
+	do {
+		announcementPageExists = false;
+		const announcementResponse = await serviceClient.fetch(
+			`api/v1/announcements?context_codes[]=course_${courseId}&start_date=${StartDate}&end_date=${endDate}&page=${announcementIncrement}&per_page=${perPage}`
+		);
+		const tempAnnouncementLink = announcementResponse.headers.map.link.split(',');
 
-  do {
-    announcementPageExists = false;
-    const announcementResponse = await serviceClient.fetch(
-      `api/v1/announcements?context_codes[]=course_${courseId}&start_date=${StartDate}&end_date=${endDate}&page=${announcementIncrement}&per_page=${perPage}`
-    );
-    announcementIncrement++;
-    const tempAnnouncementLink =
-      announcementResponse.headers.map.link.split(",");
+		if (announcementResponse.ok) {
```

Comment:
Better would be to just check `!ok`  and immediately throw an error.
```js
if (!announcementResponse.ok) {
    throw new Error(`Enrollments sync failed ${enrollmentResponse.status}:${enrollmentResponse.statusText}.`);
}

const announcement = await announcementResponse.json();
let enrollmentIncrement = 1;
let enrollmentPageExists;
....etc.
```

Then you don't have to put the whole block into two branches of if/else.
For me, it is much more readable.